src/legacy/server/index_patterns/mixin.ts

```diff
@@ -50,10 +49,8 @@ export function indexPatternsMixin(kbnServer: KbnServer, server: Legacy.Server)
     ) => callWithRequest(request, endpoint, params, options);
     return server.indexPatternsServiceFactory({ callCluster });
   });
-
-  registerRoutes(kbnServer.newPlatform.setup.core);
```

Comment:
Why was this removed?

If it's not needed - Then `src/plugins/data/server/index_patterns/routes.ts` is obsolete.
If it's needed, then it needs to be restored.

Reply:
`registerRoutes` is moved to `src/data/server`. So, it should be removed. If not, we're registering same server path twice. 

Without `registerRoutes`, `mixin.ts` is still necessary because `x-pack/infra` still uses `indexPatternsServiceFactory`.

I guess this file should remain for backward compatibility until 8.x.

Reply:
I guess we can remove `getIndexPatternsService` safely. But I'm not sure about backward compatibility for user plugins. So, I left it.